### PR TITLE
Doc wc_RsaPSS_Sign/Verify/CheckPadding

### PR DESCRIPTION
**Pull request by kojo1**
Add description:
wc_RsaPSS_Sign
wc_RsaPSS_Verify
wc_RsaPSS_VerifyInline
wc_RsaPSS_VerifyCheck
wc_RsaPSS_VerifyCheck_ex
wc_RsaPSS_VerifyCheckInline
wc_RsaPSS_VerifyCheckInline_ex
wc_RsaPSS_CheckPadding
wc_RsaPSS_CheckPadding_ex

### CODE REVIEW

File: doc/dox_comments/header_files/rsa.h [hunk 1]
```diff
@@ -337,6 +337,333 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+    \sa wc_RsaPSS_Verify
+*/
+WOLFSSL_API int  wc_RsaPSS_Sign(const byte* in, word32 inLen, byte* out,
+                                word32 outLen, enum wc_HashType hash, int mgf,
+                                RsaKey* key, WC_RNG* rng);
+
+/*!
+    \ingroup RSA
+
+    \brief Decrypt input singature to verify that the message was signed by key.
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:390]: "singature" should be "signature".  For reference, API doc in rsa.c is:

```
/* Verify the message signed with RSA-PSS.
 * Salt length is equal to hash length.
 *
 * in     Buffer holding encrypted data.
 * inLen  Length of data in buffer.
 * out    Pointer to address containing the PSS data.
 * hash   Hash algorithm.
 * mgf    Mask generation function.
 * key    Public RSA key.
 * returns the length of the PSS data on success and negative indicates failure.
 */
```

File: doc/dox_comments/header_files/rsa.h [hunk 2]
```diff
@@ -337,6 +337,333 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+                                  word32 outLen, enum wc_HashType hash, int mgf,
+                                  RsaKey* key);
+
+/*!
+    \ingroup RSA
+
+/*!
+    \ingroup RSA
+
+    \brief Decript input singature to verify that the message was signed by RSA key.
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:444]: "Decript" should be "Decrypt", "singature" should be "signature". For reference, API doc in rsa.c is:

```
/* Verify the message signed with RSA-PSS.
 * The input buffer is reused for the output buffer.
 * Salt length is equal to hash length.
 *
 * in     Buffer holding encrypted data.
 * inLen  Length of data in buffer.
 * out    Pointer to address containing the PSS data.
 * hash   Hash algorithm.
 * mgf    Mask generation function.
 * key    Public RSA key.
 * returns the length of the PSS data on success and negative indicates failure.
 */
```

File: doc/dox_comments/header_files/rsa.h [hunk 3]
```diff
@@ -337,6 +337,333 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+    \sa wc_RsaPSS_CheckPadding
+*/
+
+WOLFSSL_API int  wc_RsaPSS_VerifyInline(byte* in, word32 inLen, byte** out,
+                                        enum wc_HashType hash, int mgf,
+                                        RsaKey* key);
+/*!
+    \ingroup RSA
+
+    \brief Decript singature to verify that the message was signed by key with checking
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:494]: Please check argument names for correctness.  For reference, API doc in rsa.c is:

```
/* Verify the message signed with RSA-PSS.
 * Salt length is equal to hash length.
 *
 * in     Buffer holding encrypted data.
 * inLen  Length of data in buffer.
 * out    Pointer to address containing the PSS data.
 * outLen Length of the output.
 * digest Hash of the data that is being verified.
 * digestLen Length of hash.
 * hash   Hash algorithm.
 * mgf    Mask generation function.
 * key    Public RSA key.
 * returns the length of the PSS data on success and negative indicates failure.
 */
```

File: doc/dox_comments/header_files/rsa.h [hunk 4]
```diff
@@ -337,6 +337,333 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+*/
+WOLFSSL_API int  wc_RsaPSS_VerifyCheck(byte* in, word32 inLen,
+                               byte* out, word32 outLen,
+                               const byte* digest, word32 digestLen,
+                               enum wc_HashType hash, int mgf,
+                               RsaKey* key);
+/*!
+    \ingroup RSA
+
+    \brief Decript singature to verify that the message was signed by key with checking
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:554]: This description is not quite accurate.  Check argument descriptions, as there is no "outLen".  Also missing "digest" and "digestLen".  For reference, API doc in rsa.c is:

```
/* Verify the message signed with RSA-PSS.
 * The input buffer is reused for the output buffer.
 * Salt length is equal to hash length.
 *
 * in     Buffer holding encrypted data.
 * inLen  Length of data in buffer.
 * out    Pointer to address containing the PSS data.
 * digest Hash of the data that is being verified.
 * digestLen Length of hash.
 * hash   Hash algorithm.
 * mgf    Mask generation function.
 * key    Public RSA key.
 * returns the length of the PSS data on success and negative indicates failure.
 */
```

File: doc/dox_comments/header_files/rsa.h [hunk 5]
```diff
@@ -337,6 +337,333 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+*/
+WOLFSSL_API int  wc_RsaPSS_VerifyCheckInline(byte* in, word32 inLen, byte** out,
+                               const byte* digest, word32 digentLen,
+                               enum wc_HashType hash, int mgf,
+                               RsaKey* key);
+
+/*!
+    \ingroup RSA
+
+    \brief Decript singature to verify that the message was signed by key and check the padding.
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:614]: This description is inaccurate for wc_RsaPSS_CheckPadding(). We should use the API description from the comment in rsa.c:

```
/* Checks the PSS data to ensure that the signature matches.
 * Salt length is equal to hash length.
 *
 * in        Hash of the data that is being verified.
 * inSz      Length of hash.
 * sig       Buffer holding PSS data.
 * sigSz     Size of PSS data.
 * hashType  Hash algorithm.
 * returns BAD_PADDING_E when the PSS data is invalid, BAD_FUNC_ARG when
 * NULL is passed in to in or sig or inSz is not the same as the hash
 * algorithm length and 0 on success.
 */
```

**Review by cconlon — CHANGES_REQUESTED** [review]
Thanks Kojo, please see comments

File: doc/dox_comments/header_files/rsa.h (hunk 1 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h]: fixed

File: doc/dox_comments/header_files/rsa.h (hunk 2 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h]: fixed

File: doc/dox_comments/header_files/rsa.h (hunk 3 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h]: fixed

File: doc/dox_comments/header_files/rsa.h (hunk 4 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h]: fixed

File: doc/dox_comments/header_files/rsa.h (hunk 5 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h]: fixed

File: doc/dox_comments/header_files/rsa.h [hunk 6]
```diff
@@ -337,6 +337,563 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+    \sa wc_RsaPSS_Sign
+    \sa wc_RsaPSS_VerifyInline
+    \sa wc_RsaPSS_CheckPadding
+*/
+WOLFSSL_API int  wc_RsaPSS_Verify(byte* in, word32 inLen, byte* out,
+                                  word32 outLen, enum wc_HashType hash, int mgf,
+                                  RsaKey* key);
+
+/*!
+    \ingroup RSA
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:439]: Extra "\ingroup RSA" here?  Seems to be a duplicate of directly below this.

**Review by cconlon — CHANGES_REQUESTED** [review]

File: doc/dox_comments/header_files/rsa.h [hunk 7]
```diff
@@ -337,6 +337,560 @@ WOLFSSL_API int  wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out,
...
+    \sa wc_RsaPSS_VerifyCheck
+    \sa wc_RsaPSS_VerifyCheck_ex
+    \sa wc_RsaPSS_VerifyCheckInline
+    \sa wc_RsaPSS_VerifyCheckInline_ex
+    \sa wc_RsaPSS_CheckPadding
+*/
+WOLFSSL_API int  wc_RsaPSS_CheckPadding_ex(const byte* in, word32 inLen, byte* sig,
+                                        word32 sigSz,
+                                        enum wc_HashType hashType, int saltLen, int bits);
+    \ingroup RSA
```
Comment (cconlon) [inline doc/dox_comments/header_files/rsa.h:892]: Can you look at this section?  It looks like we are missing an open comment.

**Review by cconlon — CHANGES_REQUESTED** [review]

File: doc/dox_comments/header_files/rsa.h (hunk 6 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h:439]: Fixed

File: doc/dox_comments/header_files/rsa.h (hunk 7 above)
Comment (kojo1) [reply doc/dox_comments/header_files/rsa.h:893]: fixed

**Review by cconlon — APPROVED** [review]

**Pull request merged**